update exist function

SFJohnson24 opened the pull request:
[Rule (8).zip](https://github.com/user-attachments/files/16589915/Rule.8.zip)
this is from before this code was implemented but it contains the define/dataset/rule for this branch.  I do want to make sure this preserves others rules that use exist/not_exist so it would be worth checking another rule or two from this list and ensuring exists functionality is preserved with the changes: [rules_20240806_135751.xlsx](https://github.com/user-attachments/files/16589936/rules_20240806_135751.xlsx)

RamilCDISC reviewed (changes requested):
The new changes preserves the old logic while adding new functionality of checking within the lists that may appear. This is great. If you would please add a test case in unit tests that verifies the new functionality of checking in the nested lists, we can approve and close.

SFJohnson24 commented:
@RamilCDISC I updated the test suite.  In doing so, I noticed the nested logic was behaving differently than the original function which would return all False or True depending on presence of the target.  I changed the nesting logic to reflect this so we were not getting mixed returns of True/False bools (I believe the presence of True would override the other Falses though in processing which is why I believe this still behaves the same as before.  I tested on positive and negative data for CG0320.)  Would love for you to give it a second glance though but I believe the one line did not change anything except for giving a cleaner result consistent with the way the function worked previously.

RamilCDISC reviewed (approved):
The logic is successfully updated to handle the cases, where a column may exist with in the nested lists. Also the test suite is updated to include cases for the new functionality, as requested in the first review. The PR is good to be approved
